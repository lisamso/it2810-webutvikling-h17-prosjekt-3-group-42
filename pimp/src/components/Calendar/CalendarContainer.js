import React from 'react';
import {Component} from "react/cjs/react.production.min";
import Calendar from 'react-calendar/build/entry.nostyle';

class CalendarContainer extends Component {
  constructor(props) {
    super(props);
  }
  
  // Makes a specific date green
  // Can use this to mark event dates
  // The date and month must be integers
  addEventOnDay(date, eventHover) {
    const buttons = document.getElementsByClassName(
      "react-calendar__month-view__days__day");
    const currentMonth = new Date().getMonth();
    for (let i = 0; i < buttons.length; i++) {
      let btn = buttons[i]
      const btnDate = new Date(
        btn.firstChild.attributes.item(0).textContent);
      if (btnDate.getDate() === date.getDate() &&
          btnDate.getMonth() === date.getMonth() &&
          btnDate.getMonth() === currentMonth) {
        btn.style.background = "#2ecc71";
        btn.style.color = "black";
        btn.title = eventHover;
        btn.className += " tile-has-event"
      }
    }
  }
  
  // Makes all the days transparent
  deleteEvents() {
    const buttons = document.getElementsByClassName(
      "react-calendar__month-view__days__day");
    for (let i = 0; i < buttons.length; i++) {
      buttons[i].style.background = "transparent";
      const date = new Date(
        buttons[i].firstChild.attributes.item(0).textContent);
      // If saturday or sunday, make text emerald
      if (date.getDay() === 6 || date.getDay() === 0) {
        buttons[i].style.color = "#2ecc71";
      }
      // Else make text white
      else {
        buttons[i].style.color = "white";
      }
    }
  }
  
  // Runs every time the EventContainer updates
  // Deleting makes all the days transparent before it goes through
  // the events and makes any day having an event green.
  componentWillUpdate() {
    this.deleteEvents();
    JSON.parse(localStorage.getItem('events')).map((event)=>{
      let eventHover = `${event['title']}: ${event['text']}`
      this.addEventOnDay(new Date(Date.parse(event['eventAt'])), eventHover);
    })
  }
  
  render() {
    return (
      <Calendar showNavigation={false}/>
    )
  }
}

export default CalendarContainer;
