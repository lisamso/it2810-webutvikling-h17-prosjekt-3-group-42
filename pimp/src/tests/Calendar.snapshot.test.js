import CalendarContainer from '../components/Calendar/CalendarContainer';
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });



test('CalendarContainer component should render as expected', () =>{
  const component = shallow(<CalendarContainer />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
describe('Test the functions of the CalendarContainer component.', () =>{

  it('', () =>{
    const component = shallow(<CalendarContainer />);
    const tree = toJson(component);
    const c = component.find('Calendar');

  });


});
