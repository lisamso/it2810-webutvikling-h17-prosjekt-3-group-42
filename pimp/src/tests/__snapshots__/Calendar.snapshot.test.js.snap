// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarContainer component should render as expected 1`] = `
<Calendar
  maxDetail="month"
  minDetail="century"
  returnValue="start"
  showNavigation={true}
  showNeighboringMonth={true}
  view="month"
/>
`;
